zkevm-circuits/src/evm_circuit/execution/callop.rs

```diff
@@ -256,107 +301,152 @@ impl<F: Field> ExecutionGadget<F> for CallOpGadget<F> {
+                    (
+                        CallContextFieldTag::GasLeft,
+                        cb.curr.state.gas_left.expr() - gas_cost - callee_gas_left.clone(),
+                    ),
+                    (
+                        CallContextFieldTag::MemorySize,
+                        memory_expansion.next_memory_word_size(),
+                    ),
+                    (
+                        CallContextFieldTag::ReversibleWriteCounter,
+                        cb.curr.state.reversible_write_counter.expr() + 1.expr(),
+                    ),
+                ] {
+                    cb.call_context_lookup(true.expr(), None, field_tag, value);
+                }
+
+                // Setup next call's context.
+                for (field_tag, value) in [
+                    (CallContextFieldTag::CallerId, cb.curr.state.call_id.expr()),
+                    (CallContextFieldTag::TxId, tx_id.expr()),
+                    (CallContextFieldTag::Depth, depth.expr() + 1.expr()),
+                    (CallContextFieldTag::CallerAddress, caller_address),
+                    (CallContextFieldTag::CalleeAddress, callee_address),
+                    (CallContextFieldTag::CallDataOffset, cd_address.offset()),
+                    (CallContextFieldTag::CallDataLength, cd_address.length()),
+                    (CallContextFieldTag::ReturnDataOffset, rd_address.offset()),
+                    (CallContextFieldTag::ReturnDataLength, rd_address.length()),
+                    (
+                        CallContextFieldTag::Value,
+                        select::expr(is_delegatecall.expr(), current_value.expr(), value.expr()),
+                    ),
+                    (CallContextFieldTag::IsSuccess, is_success.expr()),
+                    (CallContextFieldTag::IsStatic, is_staticcall.expr()),
+                    (CallContextFieldTag::LastCalleeId, 0.expr()),
+                    (CallContextFieldTag::LastCalleeReturnDataOffset, 0.expr()),
+                    (CallContextFieldTag::LastCalleeReturnDataLength, 0.expr()),
+                    (CallContextFieldTag::IsRoot, 0.expr()),
+                    (CallContextFieldTag::IsCreate, 0.expr()),
+                    (CallContextFieldTag::CodeHash, callee_code_hash.expr()),
+                ] {
+                    cb.call_context_lookup(
+                        true.expr(),
+                        Some(callee_call_id.expr()),
+                        field_tag,
+                        value,
+                    );
+                }
+
+                // Give gas stipend if value is not zero
+                let callee_gas_left =
+                    callee_gas_left + has_value * GAS_STIPEND_CALL_WITH_VALUE.expr();
+
+                // Both CALL and CALLCODE have an extra stack pop `value`, and
+                // opcode DELEGATECALL has two extra call context lookups - current
+                // caller address and current value.
+                let rw_counter_delta = 44.expr()
```

Comment:
Why an increment of `1`? I'm missing the RW that is done always and isn't conditional. Could you point it out?

Reply:
I think this is the extra caller balance lookup to determine if there's sufficient balance.

Reply:
> I think this is the extra caller balance lookup to determine if there's sufficient balance.

you are right !